reframe/core/schedulers/__init__.py

```diff
@@ -215,10 +214,16 @@ def emit_preamble(self):
         pass
 
     def guess_num_tasks(self):
+        num_tasks_per_node = self.num_tasks_per_node or 1
         if isinstance(self.sched_flex_alloc_tasks, int):
             if self.sched_flex_alloc_tasks <= 0:
                 raise JobError('invalid number of flex_alloc_tasks: %s' %
                                self.sched_flex_alloc_tasks)
+            elif self.sched_flex_alloc_tasks % num_tasks_per_node != 0:
+                raise JobError(
+                    'invalid number of flex_alloc_tasks: %s, it must be a '
+                    'multiple of num_tasks_per_node: %s' %
+                    (self.sched_flex_alloc_tasks, num_tasks_per_node))
```

Comment:
This fixes the typicality of producing a correct error message, but I think we should do more than that, if possible, because essentially you need to know the `num_tasks_per_node` of a flexible test.